src/components/modern/FiatValueView.js

```diff
@@ -52,14 +53,26 @@ function FiatValueView({
 }: Props) {
   const colors = useThemeColors();
 
-  if (!value) return null;
+  const formattedValue = React.useMemo(() => {
+    if (!value) {
+      return null;
+    }
+    if (typeof value === 'number') {
+      return wrapBigNumber(value);
+    }
+    return value;
+  }, [value]);
```

Comment:
I think there is no need to use `useMemo` here, as parsing BigNumber is cheap.  Pls also rename the variable, as it's not really formatted but rather more like parsed or transformed. Usually we apply `BN` suffix to such cases.
```suggestion
  const valueBN = wrapBigNumberOrNil(value);
```

Reply:
Done.